### PR TITLE
Use Content-Signature protocol from Autograph

### PR DESCRIPTION
**Pull request by leplatrem**
Changes had to made to kinto-signer in order to get it work with the verifier in PSM

/cc @mozmark and @franziskuskiefer 

r? @Natim

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -36,6 +36,22 @@ When the *source* collection metadata ``status`` is set to ``"to-sign"``, it wil
...
 
 
+Content-Signature protocol
+--------------------------
+
+Kinto-signer produces signatures for the content of Kinto collections using
+`ECDSA <https://fr.wikipedia.org/wiki/Elliptic_curve_digital_signature_algorithm>`_
+with the P-384 strength.
+
+* The content is prepended with ``Content-Signature:\x00`` prior to signing.
```
Comment (Natim) [inline README.rst]: Too bad we changed our mind on that :(

File: kinto_signer/signer/__init__.py [hunk 2]
```diff
@@ -12,10 +16,11 @@ def heartbeat(request):
...
     try:
         result = signer.sign("TEST")
-        expected = set(["signature", "hash_algorithm", "signature_encoding"])
-        if len(expected.intersection(result.keys())) != 3:
+        expected = set(EXPECTED_FIELDS)
+        if len(expected.intersection(result.keys())) != len(EXPECTED_FIELDS):
             raise ValueError("Invalid response content: %s" % result)
         return True
     except Exception as e:
+        print(e)
```
Comment (Natim) [inline kinto_signer/signer/__init__.py]: nit: remove the print since it is logged.

File: kinto_signer/signer/local_ecdsa.py [hunk 3]
```diff
@@ -81,11 +91,10 @@ def verify(self, payload, signature_bundle):
 
         public_key = self.load_public_key()
         try:
-            public_key.verify(
-                signature_bytes,
-                payload,
-                hashfunc=hashlib.sha384,
-                sigdecode=ecdsa.util.sigdecode_string)
+            public_key.verify(signature_bytes,
```
Comment (Natim) [inline kinto_signer/signer/local_ecdsa.py]: Don't you need to use the x5u certificate somewhere to validate the signature?

**Comment by Natim** [comment]
I'd like to use the x5u parameter if present to validate the ``public_key`` is valid.

File: kinto_signer/signer/local_ecdsa.py (hunk 3 above)
Comment (leplatrem) [inline kinto_signer/signer/local_ecdsa.py]: Let's open an issue that's a different story I think

File: README.rst [hunk 4]
```diff
@@ -36,6 +36,22 @@ When the *source* collection metadata ``status`` is set to ``"to-sign"``, it wil
...
+* The signature is produced using the SHA-384 hash.
```
Comment (franziskuskiefer) [inline README.rst]: The signature is produced with ECDSA on P-384 using SHA-384.

File: kinto_signer/signer/local_ecdsa.py (hunk 3 above)
Comment (Natim) [inline kinto_signer/signer/local_ecdsa.py]: Yes sure. #68

**Comment by Natim** [comment]
r+

**Pull request merged**